common/tasks/dynamic_worker_pool_scheduler.go

```diff
@@ -100,12 +119,16 @@ func (pool *DynamicWorkerPoolScheduler) TrySubmit(task Runnable) bool {
 	pool.mu.Lock()
 	if pool.runningGoroutines >= pool.limiter.Concurrency() {
 		enqueued := pool.tryEnqueueLocked(task)
+		bufferSize := pool.bufferSize
 		pool.mu.Unlock()
+		pool.recordBufferSize(bufferSize)
```

Comment:
We discussed having only the background goroutine record gauges and having separate counters for enqueues/dequeues/buffered/rejected tasks on this path.

Reply:
Is this coming in a separate PR?

Reply:
I'll do it a separate PR.